rust/hyperlane-base/src/settings/loader/deprecated_arguments.rs

```diff
@@ -0,0 +1,346 @@
+
+use config::{ConfigError, Map, Source, Value, ValueKind};
+
+/// A source for loading configuration from command line arguments.
+/// Command line argument keys are case-insensitive, and the following forms are
+/// supported:
+///
+/// * `--key=value`
+/// * `--key="value"`
+/// * `--key='value'`
+/// * `--key value`
+/// * `--key` (value is an empty string)
+#[must_use]
+#[derive(Clone, Debug, Default)]
+pub struct DeprecatedCommandLineArguments {
+    /// Optional character sequence that separates each key segment in an
+    /// environment key pattern. Consider a nested configuration such as
+    /// `redis.password`, a separator of `-` would allow an environment key
+    /// of `redis-password` to match.
+    separator: Option<String>,
+
+    /// Ignore empty env values (treat as unset).
+    ignore_empty: bool,
+
+    /// Alternate source for the environment. This can be used when you want to
+    /// test your own code using this source, without the need to change the
+    /// actual system environment variables.
+    source: Option<Vec<OsString>>,
+}
+
+#[allow(unused)]
+impl DeprecatedCommandLineArguments {
+    pub fn separator(mut self, s: &str) -> Self {
+        self.separator = Some(s.into());
+        self
+    }
+
+    pub fn ignore_empty(mut self, ignore: bool) -> Self {
+        self.ignore_empty = ignore;
+        self
+    }
+
+    pub fn source<I, S>(mut self, source: I) -> Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        self.source = Some(source.into_iter().map(|s| s.as_ref().to_owned()).collect());
+        self
+    }
+}
+
+impl Source for DeprecatedCommandLineArguments {
+    fn clone_into_box(&self) -> Box<dyn Source + Send + Sync> {
+        Box::new((*self).clone())
+    }
+
+    fn collect(&self) -> Result<Map<String, Value>, ConfigError> {
+        let mut m = Map::new();
+        let uri: String = "program argument".into();
+
+        let separator = self.separator.as_deref().unwrap_or("-");
+
+        let mut args = if let Some(source) = &self.source {
+            ArgumentParser::from_vec(source.clone())
+        } else {
+            ArgumentParser::from_env()
+        };
+
+        while let Some((key, value)) = args
+            .next()
+            .transpose()
+            .map_err(|e| ConfigError::Foreign(Box::new(e)))?
+        {
+            if self.ignore_empty && value.is_empty() {
+                continue;
+            }
+
+            let mut key = key.to_lowercase();
+
+            // If separator is given replace with `.`
+            if !separator.is_empty() && separator != "." {
+                key = key.replace(separator, ".");
+            }
+
+            if key.ends_with("interchaingaspaymaster") {
+                key = key.replace("interchaingaspaymaster", "interchainGasPaymaster");
+            } else if key.ends_with("validatorannounce") {
+                key = key.replace("validatorannounce", "validatorAnnounce");
+            }
```

Comment:
Will we be able to remove `DeprecatedCommandLineArguments` once your big configuration rework is merged? Kind of unfortunate this is needed

Reply:
No, it will have to wait until we actually stop supporting the old config format.